src/schema/utils.ts

```diff
@@ -185,3 +186,22 @@ export const isEmptyObject = (o: any) => isDeepStrictEqual(o, {})
 export function keys<A extends object>(a: A): (keyof A)[] {
   return Object.keys(a) as any
 }
+
+export function ensureDepIsInstalled(depName: string) {
+  try {
+    require(depName)
```

Comment:
We could make this check against the package.json contents.

Doing so would be stricter.

Not sure about the benefit...

It would for example make it not nice to use prisma nexus schema plugin in a framework context because the framework [will] complains when you instal graphql or @nexus/schema deps. Catch 22.

It think you found the best way to do this... just wanted to mention this.

Reply:
Yeah, interesting. Your approach would ensure that bad stuff cannot happen in prod. eg: you have graphql or nexus schema installed as dev deps locally, but it fails in prod as the dev deps aren't installed.

I don't feel comfortable doing IOs to find the package.json in the schema lib though. We'll have to recursively walk through the files again and it sucks to have to pay that price in prod

Reply:
Good point, how about:

```ts
if (process.env.NODE_ENV === 'production'){ /* expensive check */ } else { /* cheap check */ }
```

Reply:
Do you mind if we wait for the bad situations described above to happen? I'm worried about the package.json walk, even in dev. We know from the framework the kind of issues it can create to look for files, in some situations like monorepos and stuff like that. I'd like to not create a whole new category of bugs because of it. If you strongly think we should do it then I'll make the change, but my feeling is that staying simpler here is better, especially for something that might be a hard blocker

Reply:
Could use https://github.com/sindresorhus/read-pkg-up

Reply:
Seeing your comment now, yep sounds good 👍